src/Kafka/Settings/Settings.tsx

```diff
@@ -0,0 +1,164 @@
+                      </strong>
+                    </FlexItem>
+                    <FlexItem>
+                      {(() => {
+                        switch (connectionStatus) {
+                          case "On":
+                            return t(
+                              "settings.Connection_re_authentication_states.turnon"
+                            );
+                          case "TurningOff":
+                            return (
+                              <>
+                                <Spinner
+                                  size="md"
+                                  aria-valuetext={t(
+                                    "settings.Connection_re_authentication_states.turning_off"
+                                  )}
+                                  aria-label={t(
+                                    "settings.connection_re_authentication_label"
+                                  )}
+                                />{" "}
+                                {t(
+                                  "settings.Connection_re_authentication_states.turning_off"
+                                )}
+                              </>
+                            );
+                          case "Off":
+                            return t(
+                              "settings.Connection_re_authentication_states.turnoff"
+                            );
+                          case "TurningOn":
+                            return (
+                              <>
+                                <Spinner
+                                  size="md"
+                                  aria-valuetext={
+                                    "settings.Connection_re_authentication_states.turning_on"
+                                  }
+                                  aria-label={t(
+                                    "settings.connection_re_authentication_label"
+                                  )}
+                                />{" "}
+                                {t(
+                                  "settings.Connection_re_authentication_states.turning_on"
+                                )}
+                              </>
+                            );
+                          default:
+                            return null;
+                        }
+                      })()}
+                    </FlexItem>
+                  </Flex>
+                </FlexItem>
+                <FlexItem>
+                  <Switch
+                    id="Connection-re-authentication-switch"
+                    aria-label={t(
+                      "settings.connection_re_authentication_label"
+                    )}
+                    isChecked={connectionStatus === "On"}
+                    isDisabled={
+                      connectionStatus === "TurningOff" ||
+                      connectionStatus === "TurningOn"
+                    }
```

Comment:
I'm seeing inconsistent behavior here:

* When I toggle the switch to the **off** state, I see a disabled switch in the **off** state while the spinner displays, then I see an enabled switch in the **off** state.
* When I toggle the switch to the **on** state, I see a disabled switch in the **off** state while the spinner displays, then I see an enabled switch in the **on** state.

The disabled state is always displaying in the **off** state for both cases.

Instead, it would be good to display the disabled state showing the previous value while the spinner displays, then update it to the new value.